### PR TITLE
Charts

### PR DESCRIPTION
**Pull request by Caleb-Nieto**
### Feature / Bug Description
Render charts

### Solution
Recharts dependency added
Chart service added to fetch the data
New Route created (/charts)

### Notes
no notes needed

### Tickets
[WMA-149](https://cdentalcaregroup.atlassian.net/browse/WMA-149)
[WMA-148](https://cdentalcaregroup.atlassian.net/browse/WMA-148)

### Screenshots / Screencasts
zoom image
Captura de pantalla 2024-08-06 124011

[WMA-149]: https://cdentalcaregroup.atlassian.net/browse/WMA-149?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WMA-148]: https://cdentalcaregroup.atlassian.net/browse/WMA-148?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Caleb-Nieto** [comment]
image

**Review by imadiaz — APPROVED** [review]

File: src/pages/charts/components/AreasChart.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        <Tooltip
+          content={(props) => (
+            <div>
+              {props.payload?.map((item) => {
+                return (
+                  <div
+                    className="bg-card-fields md:text-sm text-xs w-52 md:w-auto text-white py-2 px-4 rounded-md shadow-lg"
+                    key={item.payload.area}
+                  >
+                    <p>Area: {item.payload.area}</p>
```
Comment (imadiaz) [inline src/pages/charts/components/AreasChart.tsx:40]: can we use constants for here?

File: src/pages/charts/components/AreasChart.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                    <p>Total cards: {item.payload.totalCards}</p>
```
Comment (imadiaz) [inline src/pages/charts/components/AreasChart.tsx:41]: also this one

File: src/pages/charts/components/CreatorsChart.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        <Tooltip
+          content={(props) => (
+            <div>
+              {props.payload?.map((item, index) => {
+                return (
+                  <div
+                    className="bg-card-fields md:text-sm text-xs w-52 md:w-auto text-white py-2 px-4 rounded-md shadow-lg"
+                    key={index}
+                  >
+                    <p>Creator: {item.payload.creator}</p>
```
Comment (imadiaz) [inline src/pages/charts/components/CreatorsChart.tsx:40]: same here

File: src/pages/charts/components/MethodologiesChart.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        <Tooltip
+          content={({ payload }) => (
+            <div>
+              {payload?.map((item, index) => (
+                <div
+                  className="bg-card-fields md:text-sm text-xs w-52 md:w-auto text-white py-2 px-4 rounded-md shadow-lg"
+                  key={index}
+                >
+                  <p>Methodology: {item.payload.methodology}</p>
+                  <p>Total cards: {item.payload.totalCards}</p>
```
Comment (imadiaz) [inline src/pages/charts/components/MethodologiesChart.tsx:79]: and here

File: src/pages/charts/components/WeeksChart.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      >
+        <CartesianGrid strokeDasharray="3 3" />
+        <XAxis dataKey="week" />
+        <YAxis />
+        <Tooltip
+          content={(props) => {
+            if (props.active && props.payload && props.payload.length) {
+              return (
+                <div className="bg-card-fields text-white py-2 px-4 rounded-md shadow-lg">
+                  <p>Year: {props.payload[0].payload.year}</p>
```
Comment (imadiaz) [inline src/pages/charts/components/WeeksChart.tsx:69]: and here

**Pull request merged**